core/app/models/spree/order/checkout.rb

```diff
@@ -112,6 +112,7 @@ def define_state_machine!
               before_transition to: :complete, do: :ensure_available_shipping_rates
             end
             before_transition to: :complete, do: :ensure_promotions_eligible
+            before_transition to: :complete, do: :ensure_store_credits_eligible
```

Comment:
Anything that adds a transition I'm not a huge fan of, will talk offline though.